tensorflow/core/kernels/cudnn_rnn_ops.cc

```diff
@@ -1109,33 +1210,46 @@ class CudnnRNNParamsToCanonical<GPUDevice, T> : public CudnnRNNKernelCommon {
     if (rnn_direction_mode() == RnnDirectionMode::kRnnBidirectional) {
       num_dirs = 2;
     }
-    const int num_params_per_layer = num_params_ / num_layers / num_dirs;
+    const int num_params_weights_per_layer =
+        num_params_weights_ / num_layers / num_dirs;
     // Number of params applied on inputs. The rest are applied on recurrent
     // hidden states.
-    const int num_params_input_state = num_params_per_layer / 2;
-    CHECK(num_params_ % (num_layers * num_dirs) == 0)
-        << "Number of params is not a multiple of num_layers * num_dirs.";
-    CHECK(num_params_per_layer % 2 == 0)
-        << "Number of params per layer is not a even number.";
-
-    CHECK(num_params_ == rnn_desc->ParamsWeightRegions().size())
-        << "Number of params mismatch. Expected " << num_params_ << ", got "
-        << rnn_desc->ParamsWeightRegions().size();
+    const int num_params_input_state = num_params_weights_per_layer / 2;
+    CHECK(num_params_weights_ % (num_layers * num_dirs) == 0)
```

Comment:
Can we change all of these `CHECK`s to `OP_REQUIRES`? We prefer to report error instead of check failures in new code.